### PR TITLE
do not switch context when deploy karmada

### LINKED ISSUE #2059: shoudn't switch context when deploy or update karmada
**What would you like to be added**:
should replace `kubectl config use-context "${CONTEXT}"` with `kubectl --context="${CONTEXT}"` .

**Why is this needed**:
1. if deploy failed, it can't ensure context switch to "HOST_CLUSTER".
2. when deploy karmada, if other operations engineer switch context, karmada will failed too.

### PR DESCRIPTION
**Pull request by wangyanzhao**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Do not switch context when deploy karmada

**Which issue(s) this PR fixes**:
Fixes #2059

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Hi @wangyanzhao, thanks for your contributions.

Please sign off refer to https://github.com/karmada-io/karmada/pull/2091/checks?check_run_id=7112607767

**Comment by RainbowMango** [comment]
@wangyanzhao Testing failed, please check.
https://github.com/karmada-io/karmada/runs/7127449610?check_suite_focus=true
Seems a typo there.

**Comment by RainbowMango** [comment]
@wangyanzhao You might need to add your email to [GitHub email list](https://github.com/settings/emails), otherwise, GitHub can't count your contribution.

**Comment by wangyanzhao** [comment]
https://github.com/karmada-io/karmada/runs/7195338380?check_suite_focus=true#step:5:1202 

@RainbowMango, it doesn't look like a code bug, can you retry this test.

**Comment by RainbowMango** [comment]
done.
I'll review it tomorrow.

**Review by RainbowMango — APPROVED** [review]
Looks good to me.
/approve 

Leave LGTM to hack reviewers.
/assign @ikaven1024 @mrlihanbo @lfbear

**Comment by ikaven1024** [comment]
Search `Switched to context` in action log, still remain some `use-context`:
https://github.com/karmada-io/karmada/runs/7445487996?check_suite_focus=true#step:4:514

**Comment by wangyanzhao** [comment]
@ikaven1024, all `use-context` in test shell had been deleted.

**Comment by ikaven1024** [comment]
After deploying, shall we change the context to karmada?

The e2e failed due to the current context is not karmada.
https://github.com/karmada-io/karmada/runs/7449544501?check_suite_focus=true#step:4:579

**Comment by wangyanzhao** [comment]
> After deploying, shall we change the context to karmada?
> 
> The e2e failed due to the current context is not karmada. https://github.com/karmada-io/karmada/runs/7449544501?check_suite_focus=true#step:4:579

We shall not switch context, I will fix this e2e test.

**Comment by ikaven1024** [comment]
> > After deploying, shall we change the context to karmada?
> > The e2e failed due to the current context is not karmada. https://github.com/karmada-io/karmada/runs/7449544501?check_suite_focus=true#step:4:579
> 
> We shall not switch context, I will fix this e2e test.

What do you think about it @RainbowMango?

**Comment by RainbowMango** [comment]
> What do you think about it @RainbowMango?

Do you mean we can't or inappropriate to update the E2E?
I haven't looked at the logs.

**Comment by ikaven1024** [comment]
> > What do you think about it @RainbowMango?
> 
> Do you mean we can't or inappropriate to update the E2E? I haven't looked at the logs.

During deploying, context cannot be change. I green with it.
After deploy, shall we change the context? If we change it, it is ready, and maybe wonderful for user to use it. I am not sure about it.

File: hack/deploy-karmada-opensearch.sh [hunk 1]
```diff
@@ -42,16 +42,15 @@ fi
 TEMP_PATH=$(mktemp -d)
 cp $HOST_CLUSTER_KUBECONFIG $TEMP_PATH/kubeconfig
```
Comment (ikaven1024) [inline hack/deploy-karmada-opensearch.sh:43]: cp a temp kubeconfig is unnecessary if we don't change it

File: hack/deploy-scheduler-estimator.sh [hunk 2]
```diff
@@ -52,8 +52,6 @@ TEMP_PATH=$(mktemp -d)
 MEMBER_CLUSTER_KUBECONFIG_NAME=`basename ${MEMBER_CLUSTER_KUBECONFIG}`
 cp -rf ${MEMBER_CLUSTER_KUBECONFIG} "${TEMP_PATH}"/${MEMBER_CLUSTER_KUBECONFIG_NAME}
```
Comment (ikaven1024) [inline hack/deploy-scheduler-estimator.sh:53]: as above

File: hack/util.sh [hunk 3]
```diff
@@ -67,15 +67,14 @@ function util::get_target_source() {
 #  - $1: package name, such as "sigs.k8s.io/controller-tools/cmd/controller-gen"
 #  - $2: package version, such as "v0.4.1"
 function util::install_tools() {
-	local package="$1"
```
Comment (ikaven1024) [inline hack/util.sh:70]: irrelevant changes

Comment (ikaven1024) [reply hack/util.sh:70]: if you think this format change is necessary, plz commit it in another pr

Comment (wangyanzhao) [reply hack/util.sh]: I have reversed this.

File: hack/deploy-karmada-opensearch.sh (hunk 1 above)
Comment (wangyanzhao) [reply hack/deploy-karmada-opensearch.sh]: done

File: hack/deploy-scheduler-estimator.sh (hunk 2 above)
Comment (wangyanzhao) [reply hack/deploy-scheduler-estimator.sh]: done

**Comment by wangyanzhao** [comment]
> > > What do you think about it @RainbowMango?
> > 
> > 
> > Do you mean we can't or inappropriate to update the E2E? I haven't looked at the logs.
> 
> During deploying, context cannot be change. I green with it. After deploy, shall we change the context? If we change it, it is ready, and maybe wonderful for user to use it. I am not sure about it.

I have a case,  we should modify our ci shells which is on host cluster because of switch to karmada context after deploying karmada.

WDYT @RainbowMango @ikaven1024

**Comment by ikaven1024** [comment]
Hi @wangyanzhao shall we move forward it? It seems still some wrong in e2e test.

**Comment by ikaven1024** [comment]
Hi @wangyanzhao i picked your commit and fixed the e2e errors: #2474

**Comment by RainbowMango** [comment]
@wangyanzhao Thanks for doing this. @ikaven1024 picked your commit to #2474, your contribution is counted.

**Comment by wangyanzhao** [comment]
Very Thank You. @ikaven1024 @RainbowMango

**Pull request closed without merging**